src/component/NodeEditor/NodeEditor.test.tsx

```diff
@@ -221,6 +224,48 @@ describe('NodeEditor >', () => {
                 </div>
             ]);
         });
+
+        it('should display "Save as.." link if node contains a switch router without a result_name', () => {
+            const wrapper = mount(
+                <NodeEditor
+                    language={getBaseLanguage(languages)}
+                    translating={false}
+                    show={true}
+                    definition={definition}
+                    node={switchNode}
+                    onUpdateAction={jest.fn()}
+                    onUpdateRouter={jest.fn()}
+                    onUpdateLocalizations={jest.fn()}
+                    ComponentMap={ComponentMap}
+                />
+            );
+
+            expect(getSpecWrapper(wrapper, 'name-field').text()).toBe('Save as..');
+        });
+
+        it('should display result name field if node contains a switch router with a result_name', () => {
+            const wrapper = mount(
+                <NodeEditor
+                    language={getBaseLanguage(languages)}
+                    translating={false}
+                    show={true}
+                    definition={definition}
+                    node={waitNode}
+                    onUpdateAction={jest.fn()}
+                    onUpdateRouter={jest.fn()}
+                    onUpdateLocalizations={jest.fn()}
+                    ComponentMap={ComponentMap}
+                />
+            );
+
+            expect(getSpecWrapper(wrapper, 'name-field').prop('value')).toBe(
+                waitNode.router.result_name
+            );
+
+            wrapper.setState({ resultName: '' });
+
+            expect(getSpecWrapper(wrapper, 'name-field').prop('value')).toBe('');
```

Comment:
is this ensuring that it is visible?

Reply:
ya the result of the `getSpecWrapper` call is the `TextInputElement` so if this doesn't throw it's both visible and it contains an empty string

Reply:
Ok, that’s helpful. Did you confirm the test failure without your fix?